mesonpy/_editable.py

```diff
@@ -304,12 +304,15 @@ def rebuild(self) -> Node:
         env[MARKER] = os.pathsep.join((env.get(MARKER, ''), self._build_path))
 
         if self._verbose or bool(env.get(VERBOSE, '')):
-            print('+ ' + ' '.join(self._build_cmd))
-            stdout = None
+            # We do not want any output if there is no work to do. Code is adapted from:
+            # https://github.com/mesonbuild/meson/blob/a35d4d368a21f4b70afa3195da4d6292a649cb4c/mesonbuild/mtest.py#L1635-L1636
+            dry_run_build_cmd = self._build_cmd + ['-n']
```

Comment:
This does not work on Windows, where the compilation command is `meson compile`, which does not accept a `-n` argument. This demonstrates that this feature needs tests.